Python/compile.c

```diff
@@ -253,25 +274,59 @@ is_jump(struct instr *i)
     return IS_JUMP_OPCODE(i->i_opcode);
 }
 
+static oparg_type
+opcode_to_oparg_type(int opcode) {
+    if (HAS_TARGET(opcode)) {
+        return JUMP_TARGET_ARG;
+    }
+    else if (HAS_CONST(opcode)) {
+        return CONST_ARG;
+    }
+    else if (HAS_ARG(opcode)) {
+        return EXPLICIT_ARG;
+    }
+    else {
+        return UNUSED_ARG;
+    }
+    Py_UNREACHABLE();
+}
+
+#ifdef Py_DEBUG
```

Comment:
`arg_type_is_compatible` is used in an assert, so this needs to be `#ifndef NDEBUG`. We had this issue before IIRC.